python/lsst/ctrl/bps/htcondor/htcondor_service.py

```diff
@@ -497,6 +497,42 @@ def cancel(self, wms_id, pass_thru=None):
         _LOG.debug("deleted: %s; message = %s", deleted, message)
         return deleted, message
 
+    def ping(self, pass_thru):
+        """Check whether WMS services are up, reachable, and can authenticate
+        if authentication is required.
+
+        The services to be checked are those needed for submit, report, cancel,
+        restart, but ping cannot guarantee whether jobs would actually run
+        successfully.
+
+        Parameters
+        ----------
+        pass_thru : `str`, optional
+            Information to pass through to WMS.
+
+        Returns
+        -------
+        status : `int`
+            0 for success, non-zero for failure.
+        message : `str`
+            Any message from WMS (e.g., error details).
+        """
+        coll = htcondor.Collector()
+        secman = htcondor.SecMan()
+        status = 0
+        message = ""
+        _LOG.info("Not verifying that compute resources exist.")
```

Comment:
Does other WMSs, e.g. PanDA, verify that compute resource exists that we need this disclaimer here?

Reply:
I added this disclaimer here because of the separate, currently manual, allocation of compute resources.  I did not want someone to think that a successful ping meant their jobs should be running.